proposals/NNNN-Avoid_custom_button_subscription_when_HMI_does_not_support.md

```diff
@@ -0,0 +1,74 @@
+
+# Avoid Custom button subscription in case HMI incompatibility
+
+* Proposal: [SDL-NNNN](NNNN-Avoid_custom_button_subscription_when_HMI_does_not_support.md)
+* Author: [Getmanets Irina](https://github.com/GetmanetsIrina)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+In case HMI does not support `CUSTOM_BUTTON`, SDL should not try to subscribe an application to it.
+
+## Motivation
+
+SDL does not check hmi_capabilities before subscription to `CUSTOM_BUTTON`, so it tries to subscribe each registered application to `CUSTOM_BUTTON` even if it is unsupported by HMI.
+
+SDL should check hmi_capabilities and subscribe application to `CUSTOM_BUTTON` if it is **supported by HMI only**.
+
+## Proposed solution
+
+On the application registration, SDL should check hmi_capabilities.
```

Comment:
@GetmanetsIrina I would propose to check hmi_capabilities earlier. Just on SDL start - as at that moment SDL exactly checks what is supported by hmi. `CUSTOM_BUTTON` support can be checked at this moment as well as its state can be saved for further usage. 
Then we don't need to check hmi capabilities on each app registration. We just need to get saved state.

Reply:
@AByzhynar , updated. Please find update in dce4f393de9cda0d983a1360974136468b6b3a67